src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] arr;
+
+    public ArrayList() {
+        arr = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < arr.length) {
+            arr[size] = value;
+            size++;
+        } else {
+            resize();
+            arr[size] = value;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+        if (index == arr.length || size >= arr.length) {
+            resize();
+        }
```

Comment:
The condition 'index == arr.length' is not necessary since 'size >= arr.length' covers the scenario where the array needs resizing.